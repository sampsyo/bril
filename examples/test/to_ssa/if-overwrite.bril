@main() {
.entry:
    cond: bool = const true;
    # This is overwritten in both the `.true` and `.false` branches,
    # so should not be passed to the phi in `.zexit`.
    b: int = const 42;
    br cond .true .false;
.true:
    b: int = const 1;
    jmp .zexit;
.false:
    b: int = const 0;
    jmp .zexit;
.zexit:
    print b;
}
