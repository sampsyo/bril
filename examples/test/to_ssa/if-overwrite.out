@main {
.entry:
  cond.0: bool = const true;
  b.0: int = const 42;
  br cond.0 .true .false;
.true:
  b.2: int = const 1;
  jmp .zexit;
.false:
  b.1: int = const 0;
  jmp .zexit;
.zexit:
  b.3: int = phi b.1 b.2 .false .true;
  print b.3;
  ret;
}
